flutter-studio/src/io/flutter/profiler/Memory.java

```diff
@@ -111,354 +116,365 @@ public Object getValueAt(int rowIndex, int columnIndex) {
+        addInstanceToTreeModel(node);
 
-      // TODO(terry): Fake node for now need to hookup Instance Detail node.
-      DefaultMutableTreeNode instanceDetailNode = new DefaultMutableTreeNode();
-      addDetailNodeInstance(node, instanceDetailNode);
-    }
+        // TODO(terry): Fake node for now need to hookup Instance Detail node.
+        DefaultMutableTreeNode instanceDetailNode = new DefaultMutableTreeNode();
+        addDetailNodeInstance(node, instanceDetailNode);
+      }
 
-    instanceObjects.setModel(_myInstancesTreeModel);
+      instanceObjects.setModel(_myInstancesTreeModel);
 
-    DefaultMutableTreeNode r = TreeUtil.deepCopyTree(_myInstancesRoot, (DefaultMutableTreeNode)_myInstancesRoot.clone());
-    //TreeUtil.sortClasses(_myInstancesRoot);
+      DefaultMutableTreeNode r = TreeUtil.deepCopyTree(_myInstancesRoot, (DefaultMutableTreeNode)_myInstancesRoot.clone());
+      //TreeUtil.sortClasses(_myInstancesRoot);
 
-    view.updateClassesStatus(instances.size() + " Instances loaded.");
-    // TODO(terry): enabling clicking in classes again.
+      view.updateClassesStatus(instances.size() + " Instances loaded.");
+      // TODO(terry): enabling clicking in classes again.
 
-    // Update the new data.
-    _myInstancesTreeModel.reload();
-  }
-}
-
-final class TreeUtil {
-  private static boolean ascending;
-  private static Comparator<DefaultMutableTreeNode> tnc = Comparator.comparing(DefaultMutableTreeNode::isLeaf)
-    .thenComparing(n -> n.getUserObject().toString());
-
-  private static Comparator<DefaultMutableTreeNode> tncCount =
-    Comparator.comparing(DefaultMutableTreeNode::isLeaf)
-      .thenComparing(n -> ((Memory.ClassNode)(n.getUserObject())).getInstancesCount());
-
-  private static void timsort(DefaultMutableTreeNode parent) {
-    int n = parent.getChildCount();
-    List<DefaultMutableTreeNode> children = new ArrayList<>(n);
-    for (int i = 0; i < n; i++) {
-      children.add((DefaultMutableTreeNode)parent.getChildAt(i));
+      // Update the new data.
+      _myInstancesTreeModel.reload();
     }
+  }
 
-    Collections.sort(children, tncCount);
+  final class TreeUtil {
+    private static boolean ascending;
+    private static Comparator<DefaultMutableTreeNode> tnc = Comparator.comparing(DefaultMutableTreeNode::isLeaf)
+      .thenComparing(n -> n.getUserObject().toString());
 
-    parent.removeAllChildren();
-    if (ascending) {
-      children.forEach(parent::add);
-    }
-    else {
-      int numChildren = children.size();
-      for (int idx = numChildren - 1; idx >= 0; idx--) {
-        parent.add(children.get(idx));
+    private static Comparator<DefaultMutableTreeNode> tncCount =
+      Comparator.comparing(DefaultMutableTreeNode::isLeaf)
+        .thenComparing(n -> ((Memory.ClassNode)(n.getUserObject())).getInstancesCount());
+
+    private static void timsort(DefaultMutableTreeNode parent) {
```

Comment:
what does timsort mean?

Reply:
TimSort is an insertion and merge sort.  One of the fastest sort algorithms around see [TimSort](https://dev.to/s_awdesh/timsort-fastest-sorting-algorithm-for-real-world-problems--2jhd). Used in Java and Python too.